docs/DeveloperGuide.md

```diff
@@ -671,29 +673,416 @@ testers are expected to do more *exploratory* testing.
    1.  Re-launch the app by double-clicking the jar file.<br>
        Expected: The newest order added is retained.
 
-1. _{ more test cases …​ }_
+
+### Adding a customer
+
+**Note:** Some of the test cases may depend on previous test cases, especially those on testing customers with duplicate/similar names. You are advised to follow the test cases in order. <br> 
+
+**Tips:** All the prerequisites below will be fulfilled if you start off with the default sample data and follow the test cases in sequence.
+
+1. Adding a unique customer with all parameters specified.
+
+    1. Prerequisites: Customer with name `John Doe` or other similar name does not already exist in the address book.
+
+    2. Test case: `addcustomer n/John Doe p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 t/friends t/owesMoney`<br>
+       Expected: The customer is successfully added. Details of the added customer shown in the status message.
+
+2. Adding a unique customer with all parameters specified using command aliases.
+
+    1. Prerequisites: Customer with name `Betsy Crowe` or other similar name does not already exist in the address book.
+   
+    2. Test case: `addc n/Betsy Crowe t/friend e/betsycrowe@example.com a/Newgate Prison p/1234567 t/criminal` <br>
+       Expected: The customer is successfully added. Details of the added customer shown in the status message.
+   
+3. Adding a customer with exact same name.
+
+   1. Prerequisites: Customer with name `Betsy Crowe` already exist in the address book.
+
+   2. Test case: `addcustomer n/Betsy Crowe t/friend e/betsycrowe@duplicate.com a/Newgate Prison p/12345678 t/criminal` <br>
+      Expected: No customer is added. Error details shown in the status message. Status bar remains the same.
+
+4. Adding a customer with a [similar name](#similar) and without the optional `tag` field.
+
+    1. Prerequisites: Customer with name `Betsy Crowe` but not `Betsy crowe` already exist in the address book.
```

Comment:
```suggestion
    1. Prerequisites: Customer with a name containing `Betsy Crowe` but not `Betsy Crowe` already exist in the address book.
```

Reply:
This is wrong btw, the `Betsy crowe` is correct

Reply:
LOL. What a savage